appinventor/buildserver/src/com/google/appinventor/buildserver/tasks/android/XmlConfig.java

```diff
@@ -160,6 +170,26 @@ private boolean createAnimationXml(File animDir) {
     return true;
   }
 
+  /**
+   * Create the custom xml files for the app.
+   */
+  private boolean createXmls(Map<String, Set<String>> xmlsNeeded) {
+    for (Map.Entry<String, Set<String>> component : xmlsNeeded.entrySet()) {
+      for (String xml : component.getValue()) {
+        String[] parts = xml.split(":", 2);
+        context.getReporter().log("Creating " + parts[0]);
+        File file =
+            new File(
+                createDir(context.getPaths().getResDir(), new File(parts[0]).getParent()),
+                new File(parts[0]).getName());
```

Comment:
We should probably do some validation of `parts[0]` here. My concern is that a maliciously crafted components.json might have something like "../../../../../aapt" or similar in an attempt to overwrite executable files outside of the build directory.

Reply:
I think we can't overwrite other files than XML because the file extension is hardcoded. UsesXmls only specifies the file name. But I assumed that the file will be saved only in a single folder nesting. I haven't tested it with a larger nesting. Maybe we need to validate by checking the first folder names and limit to a few hardcoded names like xml, drawable etc.

Reply:
My point is that if you edit the JSON file you can make the filename be something else since the .xml is added at the point of the component processing when the extension is built, not at app build time, which is when this code is run. It would be less of a problem if you added the ".xml" here rather than in the component processor (which would be behavior that matches your statement).

Reply:
Ok, I've researched and tested a few things. We can create any folder but it won't be included in the android resources. We have to use android defined folders. A folder can't contain a subfolder. Names can contain lowercase and uppercase letters, numbers, underscore, dash and plus (including configuration), can't start with numbers. I will use this to return an error for the file or folder name if it is outside of this range. 

As for the .xml extension, I think to check it, if the file contains the .xml extension, we can leave it, if it does not contain the extension, I will add it. If it contains another extension, return an error or change it to xml?